share/vss/vss.go

```diff
@@ -0,0 +1,628 @@
+// in "Provably Secure Distributed Schnorr Signatures and a (t, n) Threshold
+// Scheme for Implicit Certificates".
+// VSS enables a dealer to share a secret securely and verifiably with a list of
+// verifiers where at most t-1 out of n participants (dealer and/or verifiers)
+// are allowed to be malicious. The verifiability of the process prevents a
+// malicious dealer from influencing the outcome to his advantage as each
+// verifier can check the validity of the received share. The protocol has the
+// following steps:
+//
+//   1) The dealer sends a Deal to every verifier using `Deals()`. Each deal must
+//    be sent securely to one verifier whose public key is at the same index than
+//    the index of the Deal.
+//   2) Each verifier processes the Deal with `ProcessDeal`.
+//    This function returns a Response which can be twofold:
+//   - an approval, to confirm a correct deal
+//   - a complaint to announce an incorrect deal notifying others that the
+//     dealer might be malicious.
+//    All Responses must be broadcasted to every verifiers and the dealer.
+//   3) The dealer can respond to each complaint by a justification revealing the
+//    share he originally sent out to the accusing verifier. This is done by
+//    calling `ProcessResponse` on the `Dealer`.
+//   4) The verifiers refuse the shared secret and abort the protocol if there
+//    are at least t complaints OR if a Justification is wrong. The verifiers
+//    accept the shared secret if there are at least t approvals at which point
+//    any t out of n verifiers can reveal their shares to reconstruct the shared
+//    secret.
+package vss
+
+import (
+	"bytes"
+	"crypto/cipher"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/share"
+	"github.com/dedis/crypto/sign"
+)
+
+// Dealer encapsulates for creating and distributing the shares and for
+// replying to any Responses.
+type Dealer struct {
+	suite  abstract.Suite
+	reader cipher.Stream
+	// long is the longterm key of the Dealer
+	long          abstract.Scalar
+	pub           abstract.Point
+	secret        abstract.Scalar
+	secretCommits []abstract.Point
+	verifiers     []abstract.Point
+	// threshold of shares that is needed to reconstruct the secret
+	t int
+	// sessionID is a unique identifier for the whole session of the scheme
+	sessionID []byte
+	// list of deals this Dealer has generated
+	deals []*Deal
+	*aggregator
+}
+
+// Deal encapsulates the verifiable secret share and is sent by the dealer to a verifier.
+type Deal struct {
+	// SessionID is a unique session identifier for this protocol run
+	SessionID []byte
+	// SecShare is the private share generated by the dealer
+	SecShare *share.PriShare
+	// RndShare is the random share generated by the dealer
+	RndShare *share.PriShare
+	// T is the treshold used for this secret sharing run
+	T uint32
+	// Commitments are the coefficients used to verify the shares against.
```

Comment:
... Please let us resist to the temptation of not liberate our inner Torvalds, otherwise nothing will gets ever merged. If you all do, I'll do too and it'll super annoying ^^

In any case, there are comments which form "sentence" and comments which are not sentences, ex: `list of deals this dealer has generated`. 
I don't see any reasons to put a point on non-sentences comments. Do you ?